src/realm/sync/config.hpp

```diff
@@ -224,6 +224,9 @@ struct SyncConfig {
 
     bool simulate_integration_error = false;
 
+    std::function<void(std::shared_ptr<Realm>)> subscription_initializer{nullptr};
```

Comment:
I think this (and the bool) should be a parameter in AsyncOpenTask::start() similar to the other callback since it only makes sense for async open.